File: base/backend/internal/models/login_log.go
package models

type LoginLog struct {
	BaseModel
	TenantID uint64 `gorm:"index;comment:租户ID" json:"tenantId"`
	UserID   uint64 `gorm:"index;comment:用户ID" json:"userId"`
	Username string `gorm:"size:64;comment:用户名" json:"username"`
	IP       string `gorm:"size:64;comment:IP" json:"ip"`
	Agent    string `gorm:"size:512;comment:UserAgent" json:"agent"`
	Status   int    `gorm:"comment:状态 1成功 0失败" json:"status"`
	Message  string `gorm:"size:256;comment:消息" json:"message"`
}

func (LoginLog) TableName() string {
	return "base_login_log"
}

File: base/frontend/src/api/login-log.ts
import request from '@/utils/request'

export interface LoginLog {
  id: number
  tenantId: number
  userId: number
  username: string
  ip: string
  agent: string
  status: number
  message: string
  createdAt: string
}

export interface LoginLogQuery {
  page: number
  size: number
  username?: string
  status?: number
  startAt?: string
  endAt?: string
}

export function getLoginLogList(params: LoginLogQuery) {
  return request.get('/login-logs', { params })
}

export function deleteLoginLogs(ids: number[]) {
  return request.post('/login-logs/delete', { ids })
}

export function clearLoginLogs(days: number) {
  return request.post('/login-logs/clear', null, { params: { days } })
}

export function exportLoginLogs(params: Omit<LoginLogQuery, 'page' | 'size'>) {
  return request.get('/login-logs/export', {
    params,
    responseType: 'blob'
  })
}
